extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -94,6 +118,37 @@ class AmpYoutube extends AMP.BaseElement {
   }
 
   /** @private */
+  getPlayerReadyPromise_() {
+    return new Promise(resolve => {
+      this.getWin().addEventListener('message', event => {
+        const data = JSON.parse(event.data);
+        if (event.origin != 'https://www.youtube.com' ||
+            event.source != this.iframe_.contentWindow) {
+          return;
+        }
+        if (data.event == 'onReady') {
+          resolve(this.iframe_);
+        } else if (data.event == 'infoDelivery') {
```

Comment:
Should we separate this into two `message` listeners instead? One that handles the `onReady` and remove the listener after receiving `onReady` and the other handles the rest of messaging events?